_posts/2019-06-03-phoenix-connector.md

```diff
@@ -0,0 +1,119 @@
+
+```
+connector.name=phoenix
+phoenix.connection-url=jdbc:phoenix:host1,host2,host3:2181:/hbase
+phoenix.config.resources=/path/to/hbase-site.xml
+```
+
+The `phoenix.connection-url` is the standard Phoenix connection string, which contains the zookeeper quorum host information and root zookeeper node.
+
+The `phoenix.config.resources` is a comma separated list of configuration files, used to specify any [custom connection properties](https://phoenix.apache.org/tuning.html).
+
+# Schema
+For the most part, data types in Phoenix match up with those in Presto, with a few [minor exceptions](https://prestosql.io/docs/current/connector/phoenix.html#data-types).  One thing to note, however, is that tables in Phoenix require a primary key, whereas Presto has no concept of primary keys.  To handle this, the Phoenix connector uses a table property to specify the primary key.  For example, consider the following statement in Phoenix:
+
+```sql
+CREATE TABLE example (
+  pk_part_1 varchar,
+  pk_part_2 varchar,
+  val bigint
+  CONSTRAINT pk PRIMARY KEY (pk_part_1, pk_part_2)
+)
+```
+The equivalent statement in Presto would look something like:
+
+```sql
+CREATE TABLE phoenix.default.example (
+  pk_part_1 varchar,
+  pk_part_2 varchar,
+  val bigint
+)
+WITH (
+  rowkeys = 'pk_part_1,pk_part2'
+)
+```
+
+Additional Phoenix and HBase table properties can be specified in a [similar way](https://prestosql.io/docs/current/connector/phoenix.html#table-properties-phoenix).  Note also that the default (empty) schema in Phoenix will always map to a Presto schema named "default".
+
+# Beyond MapReduce
+When Phoenix users want to run long-running queries that scan over all/most of the data in a table, they typically have used the Phoenix [MapReduce integration](https://phoenix.apache.org/phoenix_mr.html).  However, this has limitations, as the document states:
+
+>Note: The SELECT query must not perform any aggregation or use DISTINCT as these are not supported by our map-reduce integration.
+
+This is because the framework only constructs simple Mappers which scan over each region.  To do more complex operations like aggregations, the framework would need Reducers as well.  Someone could implement that, but then they would essentially be on the path towards rewriting Hive from scratch.
+
+Presto now provides the ability to do these more complex operations.  The Phoenix connector performs the same filtered scans as the MapReduce framework, but now the Presto engine does the aggregations, joins, etc.
+
+# Federation
+With the Phoenix connector, querying multiple Phoenix clusters is as easy as querying the respective catalogs.  As a simple example, suppose we have one cluster in region `us-west1` and another cluster in `us-west2`.  If we create two catalog files, `phoenix_west.properties` and `phoenix_east.properties`, then we can query both:
```

Comment:
Shouldn't the regions be named `us-west` and `us-east` for consistency?

Reply:
Thanks for catching the typo